starters/next-react-query-tailwind/src/components/Greeting/Greeting.test.tsx

```diff
@@ -1,17 +1,42 @@
 import { render, screen } from '@testing-library/react';
-import { setupMswServer } from '../../msw/mswServer';
-import { createQueryProvider } from '../../utils/mockQueryClient';
+import { QueryClient, QueryClientProvider } from 'react-query';
 import { Greeting } from './Greeting.data';
+import { setupServer } from 'msw/node';
+import { rest } from 'msw';
 
-setupMswServer();
+const MOCK_HELLO_MESSAGE = 'Test Message';
+
+const server = setupServer(
+  rest.get('https://api.starter.dev/hello', (req, res, ctx) => {
+    return res(ctx.text(MOCK_HELLO_MESSAGE))
+  }),
+)
 
 describe('Greeting', () => {
-  it('fetches greeting from /api/hello and renders it in a heading', async () => {
-    render(<Greeting />, {
-      wrapper: createQueryProvider(),
-    });
+  beforeAll(() => server.listen())
+  afterEach(() => server.resetHandlers())
+  afterAll(() => server.close())
+
+  it('should show loading state before fetched the data from api', async () => {
```

Comment:
I don't think these tests are quite accurate or really telling what they're doing. This first test should be saying - when the API is still fetching and the data has loaded, show a loading state and the 2nd test should be saying - after the data loads, we should see the data. There should also be a test for the error case. If I were to use the same setup for both tests, I would expect both sets of tests to pass currently but they shouldn't. One should fail and one should pass.